### PR TITLE
Embed dialog: correct URL, hideControls toggle, hideControls works

### PR DESCRIPTION
**Pull request by MarcelGerber**
Notion: https://www.notion.so/owid/Embed-code-missing-url-and-hideControls-param-d08ae86bd229421ba897afc9b2a944b9

This fixes 3 issues we had:
* the `hideControls=true` param (as part of the patch) works
* the embed dialog shows the correct URL, including the base URL
* the "Hide controls" toggle at the bottom of the embed dialog is restored (it's still not integrated in a super clean way, but I think it's better than before at least, and more extensible)

image

### CODE REVIEW

File: explorer/Explorer.tsx
Comment (breck7) [inline explorer/Explorer.tsx:67]: I moved these things to explorerProgram after our convo last week. hideControls should go on explorerProgram as well. My branch isn't merged in yet, so no biggie, just wanted to mention it.

Comment (MarcelGerber) [reply explorer/Explorer.tsx:67]: Alright, I'll wait for your merge then until I merge this in.

File: grapher/controls/ShareMenu.tsx
Comment (breck7) [inline grapher/controls/ShareMenu.tsx:236]: I'm not sure how I feel about this. This is the same pattern I had added before, under the reasoning I wanted to keep Grapher and Explorer loosely coupled, but it's a bit hacky.

At this point I think it's likely that Grapher and Explorer merge, so maybe we move the hide controls component itself here?

I'm doing an attempt at fixing the hackiness in our Grapher params once and for all on my branch right now, and maybe after that it will be clear what the best thing to do is.

**Review by breck7 — CHANGES_REQUESTED** [review]
Not really "requesting changes" yet, just think we should wait a sec on this one b/c I might have some better feedback via the branch i'm on now.

File: grapher/controls/ShareMenu.tsx
Comment (MarcelGerber) [reply grapher/controls/ShareMenu.tsx:236]: Great! I agree, it does feel hacky to just pass this around to be displayed here, but I didn't really see a better way.
I'm certainly interested what you could come up with in your PR, and will wait with this PR until yours is in.

**Comment by MarcelGerber** [comment]
By @breck7's judgement, this was supposed to wait for his PR, which I think would have revamped URL params (https://www.notion.so/owid/After-changing-an-explorer-to-log-scale-url-doesn-t-revert-when-you-change-it-back-to-linear-e0ff0bf4b44e4877bbb0534ae92ce381).

Since this is not happening now, it's probably best to have this reviewed again?

**Review by danielgavrilov — APPROVED** [review]

**Pull request merged**
